Sources/SwiftDocC/Semantics/Metadata/CallToAction.swift

```diff
@@ -0,0 +1,169 @@
+                range: originalMarkup.range,
+                identifier: "org.swift.docc.\(CallToAction.self).missingLink",
+                summary: "\(CallToAction.directiveName.singleQuoted) directive requires `url` or `file` argument",
+                explanation: "The Call to Action requires a link to direct the user to."
+            )))
+
+            isValid = false
+        } else if self.url != nil && self.file != nil {
+            problems.append(.init(diagnostic: .init(
+                source: source,
+                severity: .error,
+                range: originalMarkup.range,
+                identifier: "org.swift.docc.\(CallToAction.self).tooManyLinks",
+                summary: "\(CallToAction.directiveName.singleQuoted) directive requires only one of `url` or `file`",
+                explanation: "Both the `url` and `file` arguments specify the link in the heading; specifying both of them creates ambiguity in where the call should link."
+            )))
+
+            isValid = false
+        }
+
+        if self.purpose == nil && self.label == nil {
+            problems.append(.init(diagnostic: .init(
+                source: source,
+                severity: .error,
+                range: originalMarkup.range,
+                identifier: "org.swift.docc.\(CallToAction.self).missingLabel",
+                summary: "\(CallToAction.directiveName.singleQuoted) directive requires `purpose` or `label` argument",
+                explanation: "Without a `purpose` or `label`, the Call to Action has no label to apply to the link."
+            )))
+
+            isValid = false
+        }
+
+        if let file = self.file {
+            if context.resolveAsset(named: file.url.lastPathComponent, in: bundle.rootReference) == nil {
+                problems.append(.init(
+                    diagnostic: Diagnostic(
+                        source: url,
+                        severity: .warning,
+                        range: originalMarkup.range,
+                        identifier: "org.swift.docc.Project.ProjectFilesNotFound",
+                        summary: "\(file.path) file reference not found in \(CallToAction.directiveName.singleQuoted) directive"),
+                    possibleSolutions: [
+                        Solution(summary: "Copy the referenced file into the documentation bundle directory", replacements: [])
+                    ]
+                ))
+            } else {
+                self.file = ResourceReference(bundleIdentifier: file.bundleIdentifier, path: file.url.lastPathComponent)
+            }
+        }
+
+        return isValid
+    }
+
+    public let originalMarkup: Markdown.BlockDirective
+
+    @available(*, deprecated, message: "Do not call directly. Required for 'AutomaticDirectiveConvertible'.")
+    init(originalMarkup: Markdown.BlockDirective) {
+        self.originalMarkup = originalMarkup
+    }
+}
+
+extension CallToAction.Purpose {
+    public func defaultLabel() -> String {
```

Comment:
Nit: This could also be a computed property

```suggestion
    public var defaultLabel: String {
```